src/main/java/com/xpanxion/java/assignments/student5/Worker5.java

```diff
@@ -73,4 +71,25 @@ public void ex6() {
 
         System.out.println(cats);
     }
+
+    public void ex7() {
```

Comment:
This whole approach can be simplified.  Try using the `StringTokenizer` and checkout the `TreeMap`.